frontend/app/src/components/home/PrizeContent.svelte

```diff
@@ -0,0 +1,208 @@
+<script lang="ts">
+    import Button from "../Button.svelte";
+    import type { OpenChat, PrizeContent } from "openchat-client";
+    import { _ } from "svelte-i18n";
+    import Clock from "svelte-material-icons/Clock.svelte";
+    import ButtonGroup from "../ButtonGroup.svelte";
+    import { getContext } from "svelte";
+    import { Confetti } from "svelte-confetti";
+    import { rtlStore } from "../../stores/rtl";
+    import { now } from "../../stores/time";
+    import CkBtc from "../icons/CkBtc.svelte";
+    import { toastStore } from "../../stores/toast";
+    import { claimsStore } from "../../stores/claims";
+
+    const client = getContext<OpenChat>("client");
+
+    export let content: PrizeContent;
+    export let chatId: string;
+    export let messageId: bigint;
+
+    $: total = content.prizesRemaining + content.winners.length;
+    $: percentage = (content.winners.length / total) * 100;
+    $: claimedByYou = content.winners.includes(client.user.userId);
+    $: finished = $now >= Number(content.endDate);
+    $: disabled = finished || claimedByYou || content.prizesRemaining <= 0;
+    $: timeRemaining = finished
+        ? $_("prizes.finished")
+        : client.formatTimeRemaining($now, Number(content.endDate));
+    let progressWidth = 0;
+    // let source = "../assets/ckbtc_large.jpeg";
+
+    function claim() {
+        claimsStore.add(messageId);
+        client
+            .claimPrize(chatId, messageId)
+            .then((success) => {
+                if (success) {
+                    content = {
+                        ...content,
+                        winners: [...content.winners, client.user.userId],
```

Comment:
This is not right at the moment. I guess I should be using the local updates mechanism here.